src/device/usbd.c

```diff
@@ -217,7 +203,30 @@ static usbd_class_driver_t const _usbd_driver[] =
   #endif
 };
 
-enum { USBD_CLASS_DRIVER_COUNT = TU_ARRAY_SIZE(_usbd_driver) };
+enum { BUILTIN_DRIVER_COUNT = TU_ARRAY_SIZE(_usbd_driver) };
+
+// Additional class drivers implemented by application
+static usbd_class_driver_t const * _app_driver = NULL;
+static uint8_t _app_driver_count = 0;
+
+// virtually joins built-in and application drivers together.
+// Application is positioned first to allow overwriting built-in ones.
+static inline usbd_class_driver_t const * get_driver(uint8_t drvid)
+{
+  // Application drivers
+  if ( usbd_app_driver_get_cb )
```

Comment:
I prefer to test `_app_driver_count`, since only the linker knows the existence of `usbd_app_driver_get_cb`, this `if{}` won't be optimized out if `usbd_app_driver_get_cb` not exist.

Reply:
`usbd_app_driver_get_cb` allow LTO to optimize out the code, though it is much the same with _app_driver_count, gcc is smart enough to figure it out with either.

Reply:
> gcc is smart enough to figure it out with either.

Seems gcc is smarter than IAR :)

Reply:
can you try to enable LTO option on IAR, I think IAR can only be smarter than gcc, they charge premium for that

Reply:
> I think IAR can only be smarter than gcc

I've already enabled every options. Frankly their optimization is not so impressive, many times with `const` `restrict` keywords it re-read everything from the stack. 

Their advantages are functional safety, MISRA and easy to use, for example to write a ram routine instead of fiddle with linker options and copy the routine with `memcpy`, I can use simply the keyword `__ramfunc`.

Reply:
yeah, MISRA is a super nice thing to have.